internal/pkg/service/objectstorage/cloud/azure/storage.go

```diff
@@ -17,41 +18,38 @@ type AzureObjectStorageAdapter struct {
 	azStorageAccountClient *armstorage.AccountsClient
 	azBlobContainerClient  *armstorage.BlobContainersClient
 	logger                 logr.Logger
-	resourceGroup          string
+	cluster                AzureCluster
+	storageAccountName     string
 }
 
-func NewAzureStorageService(azStorageAccountClient *armstorage.AccountsClient, azBlobContainerClient *armstorage.BlobContainersClient, logger logr.Logger, resourceGroup string) AzureObjectStorageAdapter {
+func NewAzureStorageService(azStorageAccountClient *armstorage.AccountsClient, azBlobContainerClient *armstorage.BlobContainersClient, logger logr.Logger, cluster AzureCluster) AzureObjectStorageAdapter {
 	return AzureObjectStorageAdapter{
 		azStorageAccountClient: azStorageAccountClient,
 		azBlobContainerClient:  azBlobContainerClient,
 		logger:                 logger,
-		resourceGroup:          resourceGroup,
+		cluster:                cluster,
+		storageAccountName:     cluster.Name + "observability",
 	}
 }
 
 // ExistsBucket checks if the bucket exists on Azure
 // firstly, it checks if the Storage Account exists
 // then, it checks if the Blob Container exists
 func (s AzureObjectStorageAdapter) ExistsBucket(ctx context.Context, bucket *v1alpha1.Bucket) (bool, error) {
-	// Check StorageAccount on Azure
-	availability, err := s.azStorageAccountClient.CheckNameAvailability(
-		ctx,
-		armstorage.AccountCheckNameAvailabilityParameters{
-			Name: to.Ptr(bucket.Spec.Name),
-			Type: to.Ptr("Microsoft.Storage/storageAccounts"),
-		},
-		nil)
+	s.logger.Info("CHECK BUCKET EXISTENCE - Entering")
```

Comment:
I would rather you use log.FromContext here

Reply:
Could you explain ?
We have the logger in the object

Reply:
Oh then nevermind :D